mLRS/Common/hal/esp-rxclock.h

```diff
@@ -63,39 +67,66 @@ class RxClockBase
     void SetPeriod(uint16_t period_ms);
     void Reset(void);
     void disable_isr(void);
+
+  private:
+    bool initialized = false;
 };
 
-void RxClockBase::Init(uint16_t period_ms)
+
+IRAM_ATTR void RxClockBase::Init(uint16_t period_ms)
 {
     CLOCK_PERIOD_10US = period_ms * 100; // frame rate in units of 10us
     doPostReceive = false;
 
-    // initialise the timer
+    if (initialized) return; 
+
+    // Initialize the timer
+#if defined(ESP32)
+    timer0_cfg = timerBegin(0, 800, 1);  // Timer 0, APB clock is 80 Mhz | divide by 800 is 100 KHz / 10 us, count up    
+    timerAttachInterrupt(timer0_cfg, &CLOCK_IRQHandler, true);
+    timerAlarmWrite(timer0_cfg, 1, true);
+    timerAlarmEnable(timer0_cfg);
+#elif defined(ESP8266)
     timer1_attachInterrupt(CLOCK_IRQHandler); 
     timer1_enable(TIM_DIV16, TIM_EDGE, TIM_LOOP);
-    timer1_write(50); // 5 MHz (5 ticks/us - 1677721.4 us max), 50 ticks = 10us
+    timer1_write(50); //5MHz (5 ticks/us - 1677721.4 us max), 50 ticks = 10us
```

Comment:
pl keep the older formatting of the comment :)

Reply:
Done